packages/framework/get_dependencies.sh

```diff
@@ -7,33 +7,39 @@ pushd $PWD
 cd ${script_dir}
 
 function tril_genconfig_clone_or_update_repo() {
+  pushd $PWD
   git_url=$1
   sub_dir=$2
   has_submodules=$3
+  head_sha=$4
 
   if [[ -e ${sub_dir} ]] ; then
     cd ${sub_dir}
-    git pull
```

Comment:
You can't remove the `git pull` for the repos where you did not pass in `head_sha`.  Just doing a `git fetch` does not put the repo in the correct state.  There is a missing `git pull`.  I will add the commit that does this.